### PR TITLE
Added support for calling dynamic functions defined on the host

### PR DESCRIPTION
**Pull request by syrusakbary**
# Description

Added support for calling dynamic functions defined on the host.

# Review

- [ ] Add a short description of the change to the CHANGELOG.md file

### CODE REVIEW

**Comment by syrusakbary** [comment]
Note: this PR is also changing the locations of `sys` test files to the parent dir, as they were not currently being executed

**Comment by syrusakbary** [comment]
bors try

**Comment by syrusakbary** [comment]
bors try

**Comment by syrusakbary** [comment]
bors try

**Comment by syrusakbary** [comment]
bors try

**Comment by syrusakbary** [comment]
bors try

**Comment by syrusakbary** [comment]
Merging manually as the current master is red, and this adds a small and tested feature (nothing controversial)

File: tests/ignores.txt [hunk 1]
```diff
@@ -35,6 +35,8 @@ cranelift  multi_value_imports::dylib
 singlepass multi_value_imports::dylib
 singlepass multi_value_imports::dynamic
 
+# Memory load doesn't trap as expected when out out bounds in Windows
+windows+cranelift spec::memory_grow
```
Comment (Hywan) [inline tests/ignores.txt:39]: This is a hard one :-/

**Pull request merged**